pandas/core/internals/blocks.py

```diff
@@ -257,11 +257,11 @@ def make_block_same_class(self, values, placement=None, ndim=None):
             placement = self.mgr_locs
         if ndim is None:
             ndim = self.ndim
-        return make_block(values, placement=placement, ndim=ndim, klass=self.__class__)
+        return make_block(values, placement=placement, ndim=ndim, klass=type(self))
 
     def __repr__(self) -> str:
         # don't want to print out all of the items here
-        name = pprint_thing(self.__class__.__name__)
+        name = pprint_thing(type(self).__name__)
```

Comment:
pprint_thing should be unnecessarry here.  `__name__` should always be a str, right?  this is likely a py2 holdover